src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java

```diff
@@ -62,7 +63,9 @@ public CommandResult execute(Model model) throws CommandException {
         // Find the appointment by datetime
         Appointment actualAppointment = model.getAddressBook().getAppointmentList().stream()
                 .filter(apt -> apt.appointmentDatetime.equals(appointmentDatetime))
-                .filter(apt -> apt.getSeller().equals(referencedPerson) || apt.getBuyer().equals(referencedPerson))
+                .filter(apt ->
+                        apt.getSeller().equals(referencedPerson)
+                                || apt.getBuyer().equals(Optional.of(referencedPerson)))
```

Comment:
I see that a `referencedPerson` is wrapped with an `Optional` just so that it can be compared to another `Optional`. 
```suggestion
                .filter(apt ->
                        apt.getSeller().equals(referencedPerson)
                                || apt.getBuyer().map(b -> b.equals(referencedPerson)).orElse(false))
```

Reply:
Not sure if that helps with readability, though it might with performance! I'll use yours as this seems to run quite often.